src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Metrics/ConnectionMetrics.cs

```diff
@@ -41,6 +43,11 @@ private TagList GetTags()
                 tags.Add("port", _portTag);
             }
 
+            if (_socketAddress is not null)
+            {
+                tags.Add("socket.address", _socketAddress);
+            }
```

Comment:
If available, I don't see why not to log this information also on connection metrics. With the name, I'm following [OTel recommendation](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#metric-httpclientduration) proactively using dot notation with the assumption that we will also change the instrument names by midnight of 17th.

@davidfowl @noahfalk any objections?